tests/composite_pk/test_models.py

```diff
@@ -155,3 +155,9 @@ def test_permissions(self):
         self.assertEqual(4, token.permission_set.count())
         self.assertEqual(4, user.permission_set.count())
         self.assertEqual(4, comment.permission_set.count())
+
+    def test_refresh_foreign_object(self):
+        comment = Comment.objects.get(pk=self.comment_1.pk)
+        comment.__dict__.pop("user_id", None)
```

Comment:
Hi @gmaOCR, thanks for pitching in with this one. As I mentioned on the ticket, I wanted to test actually setting `None` here, not just rewinding in time to before the descriptor was accessed.

My guess is that the set of fields that are refreshed via `refresh_from_db()` don't include ForeignObject, since they've always been ForeignKeys before and never used like their own virtual field outside that context. Does that make sense?